src/v/kafka/server/queue_depth_monitor.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+#include "kafka/server/logger.h"
+#include "utils/queue_depth_control.h"
+#include "vlog.h"
+
+namespace kafka {
+
+struct qdc_monitor {
+    struct config {
+        double latency_alpha;
+        std::chrono::milliseconds max_latency;
+        size_t window_count;
```

Comment:
should we size these integers? ... uint64_t vs size_t

Reply:
I can't imagine why we would care. the value is always going to be small and positive. using size_t means that we don't need to do any casting when interacting with vectors.